### PR TITLE
Noop decorator

### LINKED ISSUE #53: `storybook-chrome-screenshot does not support "undefined".` thrown when scs used with Storyshots
Hi. I'm attempting to use both SCS and Storyshots add-ons in my production project. `storybook-chrome-screenshot` exits successfully  but `jest` was failed. The full jest log is the following:

```txt
$ NODE_ENV=test jest storyshots.js
 => Loading custom .babelrc
 FAIL  frontend/storyshots.js
  ● Test suite failed to run

    storybook-chrome-screenshot does not support "undefined".

      at Object.<anonymous> (node_modules/storybook-chrome-screenshot/lib/withScreenshot.js:20:15)
      at Object.<anonymous> (node_modules/storybook-chrome-screenshot/lib/index.js:6:24)
      at newRequire (node_modules/@storybook/addon-storyshots/dist/require_context.js:76:12)
      at evalmachine.<anonymous>:224:34
      at runWithRequireContext (node_modules/@storybook/addon-storyshots/dist/require_context.js:102:3)
      at testStorySnapshots (node_modules/@storybook/addon-storyshots/dist/index.js:120:35)
      at Object.<anonymous> (frontend/storyshots.js:27:31)
```

Are there somebody whose some workarounds?

-  .storybook/config.js

```
import { configure, addDecorator } from '@storybook/react'
import bgs from '#/storybookDecorators/backgrounds'
import { initScreenshot, withScreenshot } from 'storybook-chrome-screenshot'

import 'babel-polyfill'

function loadStories() {
  const req = require.context('../src', true, /\.stories\.jsx$/)
  req.keys().forEach(filename => req(filename))
}

addDecorator(initScreenshot())
addDecorator(bgs())
addDecorator(withScreenshot())
configure(loadStories, module)
```

- Entry point for Storyshots:

```
/* eslint-disable */
import initStoryshots from '@storybook/addon-storyshots'

// Workaround for https://github.com/facebook/react/issues/11565#issuecomment-368877149
const ReactDOM = require('react-dom')
const React = require('react')
ReactDOM.createPortal = node => React.createElement('portal-dummy', null, node)

initStoryshots()
```

### PR DESCRIPTION
**Pull request by Quramy**
It closes #53 .

## What I did

If `getStorybookenv()` doesn't return value(e.g. jsdom and jest env), scs decorators(`init` / `with`) get  "noop" decorators.
If `getStorybookenv()` returns unknown env value(e.g. "reactNative" or "Polymer"), scs throws exception.

And I added an example using both storyshots and scs via create-react-app. You can confirm Storyshots works with:

```
cd examples/react-storyshots
yarn --pure-lockfile
yarn test # run Storyshots
yarn screenshots # run scs
```

### CODE REVIEW

File: src/clients/noop/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function noopDecorator() {
+  return (story: Function) => story;
```
Comment (tsuyoshiwada) [inline src/clients/noop/index.ts]: The output of Storyshots is `null`. ([this line](https://github.com/tsuyoshiwada/storybook-chrome-screenshot/pull/54/files#diff-2c08c45e6ab1c4cc23b1843350885410R3))
This is due to the implementation of `noopDecorator`.

---

I think that the implementation of `noopDecorator` should be as follows, but what do you think?

```typescript
import { Story } from '../../models/story';

export function noopDecorator() {
  return (storyFn: Function, ctx: Story | undefined) => (
    ctx ? storyFn(ctx) : (context: Story) => storyFn(context)
  );
}
```

File: src/withScreenshot.ts [hunk 2]
```diff
@@ -11,18 +12,22 @@ export interface WithScreenshot {
...
+      break;
+    case 'angular':
+      withScreenshot = (NgWithScreenshot as WithScreenshot);
+      break;
+    case 'vue':
+      withScreenshot = (VueWithScreenshot as WithScreenshot);
+      break;
+    default:
+      throw new Error(`storybook-chrome-screenshot does not support "${storybookEnv}".`);
+  }
```
Comment (tsuyoshiwada) [inline src/withScreenshot.ts]: :+1:

File: src/initScreenshot.ts [hunk 3]
```diff
@@ -2,22 +2,27 @@ import { getStorybookEnv } from './core/utils';
...
+      break;
+    case 'angular':
+      initScreenshot = NgInitScreenshot;
+      break;
+    case 'vue':
+      initScreenshot = VueInitScreenshot;
+      break;
+    default:
+      throw new Error(`storybook-chrome-screenshot does not support "${storybookEnv}".`);
+  }
```
Comment (tsuyoshiwada) [inline src/initScreenshot.ts]: :+1:

File: examples/react-storyshots/src/__snapshots__/Storyshots.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Storyshots Container with single viewport 1`] = `null`;
```
Comment (tsuyoshiwada) [inline examples/react-storyshots/src/__snapshots__/Storyshots.test.js.snap]: maybe, should be snapshot output as follows.

```
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
```

File: src/clients/noop/index.ts (hunk 1 above)
Comment (Quramy) [reply src/clients/noop/index.ts]: oops, I missed 2nd arg `ctx: Story | undefined` 😨  You're absolutely right 😂

File: examples/react-storyshots/src/__snapshots__/Storyshots.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Storyshots Container with single viewport 1`] = `
```
Comment (tsuyoshiwada) [inline examples/react-storyshots/src/__snapshots__/Storyshots.test.js.snap]: :tada:

**Comment by tsuyoshiwada** [comment]
@QuramyReleased as `1.1.0-alpha.4` :tada:
thx :)

**Pull request merged**